include/graphblas/reference/blas3.hpp

```diff
@@ -918,6 +937,301 @@ namespace grb {
 
 	namespace internal {
 
+		template<
+			Descriptor descr,
+			class Monoid,
+			typename InputType, typename IOType,
+			typename RIT, typename CIT, typename NIT
+		>
+		RC fold_unmasked_generic(
+			IOType &x,
+			const Matrix< InputType, reference, RIT, CIT, NIT > &A,
+			const Monoid &monoid
+		) {
+			_DEBUG_PRINT( "In grb::internal::foldr_unmasked_generic( reference )\n" );
+			RC rc = SUCCESS;
+
+			if( grb::nnz( A ) == 0 ) {
+				_DEBUG_PRINT( "The input matrix is empty, nothing to compute\n" );
+				return rc;
+			}
+
+			if ( descr & descriptors::force_row_major && descr & descriptors::transpose_left ) {
+				_DEBUG_PRINT( "Masked fold with force_row_major and transpose_left is not supported\n" );
+				return RC::ILLEGAL;
+			}
+			if ( descr & descriptors::force_row_major && descr & descriptors::transpose_matrix ) {
+				_DEBUG_PRINT( "Masked fold with force_row_major and transpose_matrix is not supported\n" );
+				return RC::ILLEGAL;
```

Comment:
is it that it cannot be supported or that there currently is no implementation? (to me it seems like transpose or no can still both be supported by CRS only?)